presto-main/src/main/java/com/facebook/presto/execution/SqlTask.java

```diff
@@ -123,7 +124,8 @@ private SqlTask(
             DataSize maxBufferSize)
     {
         this.taskId = requireNonNull(taskId, "taskId is null");
-        this.taskInstanceId = UUID.randomUUID().toString();
+        this.taskInstanceUUID = UUID.randomUUID();
```

Comment:
It's a little weird to have them both. It feels like a single part (one long) should have enough entropy. Should we simply switch to a single long?

Reply:
I clicked "approve" by an incident. Let's discuss this first.

Reply:
Ideally, we should try to keep an object in wire and memory same/similar to avoid additional decoding. I suggest to use TaskInstanceId class everywhere that I described above.

Reply:
> It's a little weird to have them both. It feels like a single part (one long) should have enough entropy. Should we simply switch to a single long?

By switching to a single long we are effectively halving from where we are currently. I think for this effort we should be focusing on keeping the functionality as is. Let me know what you think.

Reply:
I wonder why do we need to have it string. Why it cannot be simply `long` everywhere?

Reply:
> I wonder why do we need to have it string. Why it cannot be simply `long` everywhere?

Might not be a bad idea, I think I thought about this earlier but saw that SqlTaskManager's getTaskInstanceId returns a string and thought might be cleaner to keep everything SqlTask related as string instead of long. Maybe I can just wrap the long from SqlTask in String.valueOf and return in SqlTaskManager.

Reply:
Regarding going from UUID to one long, I had some more thought.

I don't think we are `halving` the entropy. Let me know if the following logic makes sense.
I think math would be something like with one long we have 2^64 combinations and with two longs we have 2^128 combinations. So we are losing a factor of 2^64 in entropy and not 2 by going from 2 longs to one long.
A simpler way to think is comparing 2 bits and 4 bits. Even though the number of bits is double the combinations we can represent are 4 and 16 respectively, which makes for a factor of 4.

Reply:
Keeping two longs as discussed offline to maintain parity with the existing code.